src/DGtal/topology/ParDirCollapse.h

```diff
@@ -0,0 +1,216 @@
+/**
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+
+#pragma once
+
+/**
+ * @file ParDirCollapse.h
+ * @author Mohamad ONAYSSI (\c mohamad.onayssi@edu.esiee.fr )
+ * @author Bibiana MARTINEZ (\c bibiana.martinez@edu.esiee.fr )
+ * @author Mohamed MELLOULI (\c mohamed.mellouli@edu.esiee.fr )
+ * ESIEE Paris
+ *
+ * @author Kacper Pluta (\c kacper.pluta@esiee.fr )
+ * Laboratoire d'Informatique Gaspard-Monge - LIGM, France
+ *
+ *
+ * @date 2015/12/22
+ *
+ *
+ * This file is part of the DGtal library.
+ */
+
+#if defined(ParDirCollapse_RECURSES)
+#error Recursive header files inclusion detected in ParDirCollapse.h
+#else // defined(ParDirCollapse_RECURSES)
+/** Prevents recursive inclusion of headers. */
+#define ParDirCollapse_RECURSES
+
+#if !defined ParDirCollapse_h
+/** Prevents repeated inclusion of headers. */
+#define ParDirCollapse_h
+
+//////////////////////////////////////////////////////////////////////////////
+// Inclusions
+#include "DGtal/helpers/StdDefs.h"
+#include "DGtal/base/Common.h"
+#include "DGtal/kernel/PointVector.h"
+// Cellular grid
+#include "DGtal/topology/CubicalComplex.h"
+#include "DGtal/topology/CubicalComplexFunctions.h"
+//////////////////////////////////////////////////////////////////////////////
+
+namespace DGtal
+{
+/////////////////////////////////////////////////////////////////////////////
+// class ParDirCollapse
+/**
+ * Description of class 'ParDirCollapse' <p>
+ * \brief Aim: Implements thinning algorithm in cubical complexes.
+ * Paper: Chaussard, J. and Couprie, M., Surface Thinning in 3D Cubical Complexes,
+ * Combinatorial Image Analysis, (2009)
+ * @tparam CC cubical complex.
```

Comment:
I would suggest to be more specific. For instance, say "Any model of cubical complex, for instance a DGtal::CubicalComplex< KhalimskySpaceND< 3, int > >". Ideally, you should have a concept of cubical complex for concept checking here, but this is not mandatory for this PR.